gtda/diagrams/_utils.py

```diff
@@ -59,6 +59,15 @@ def _filter(Xs, filtered_homology_dimensions, cutoff):
     return Xf
 
 
+def _differentiate(Xs):
+    """Change coordinates from
+    (birth, death, dimension) -> (birth, persistence, dimension)
+    """
+    X_persistence = np.copy(Xs)
```

Comment:
Why is there a need to make a copy?

Reply:
Ideally, I would like it to have no side effects on the input. In particular, if you use `ListfeatureUnion`, you might want to pass a PD to several instances.